compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -1323,11 +1331,32 @@ private void populateTypeIds(BErrorType effectiveType, BLangIntersectionTypeNode
         effectiveType.typeIdSet = BTypeIdSet.from(env.enclPkg.packageID, name, true, secondaryTypeIds);
     }
 
-    private void defineErrorIntersection(BType definedType, Location pos) {
+    private void populateUndefinedErrorIntersection(BType definedType, BLangTypeDefinition typeDefinition,
+                                                    SymbolEnv env) {
+
         BIntersectionType intersectionType = (BIntersectionType) definedType;
+        BTypeSymbol alreadyDefinedErrorTypeSymbol =
+                (BTypeSymbol) symResolver.lookupSymbolInMainSpace(env,
+                                                                  names.fromString(typeDefinition.name.value));
+        BErrorType alreadyDefinedErrorType = (BErrorType) alreadyDefinedErrorTypeSymbol.type;
         BErrorType errorType = (BErrorType) intersectionType.effectiveType;
 
-        defineErrorType(errorType, pos);
+        alreadyDefinedErrorType.typeIdSet = errorType.typeIdSet;
+        alreadyDefinedErrorType.detailType = errorType.detailType;
+        alreadyDefinedErrorType.flags = errorType.flags;
+        alreadyDefinedErrorType.name = errorType.name;
+        intersectionType.effectiveType = alreadyDefinedErrorType;
+
+        addErrorTypeDefinition(alreadyDefinedErrorType, env, typeDefinition.pos);
```

Comment:
There is already a type definition for this error right? Shouldn't we update that or replace it altogether?

Reply:
No. We have only created the ErrorType and Symbol before. Hence adding the definition now